index.js

```diff
@@ -70,89 +70,91 @@ const commonOpts = {
 
 async function deleteOlderReleases(keepLatest) {
   let releaseIdsAndTags = [];
-  try {
-    let data = await fetch({
-      ...commonOpts,
-      path: `/repos/${owner}/${repo}/releases?per_page=100`,
-      method: "GET",
-    });
-    data = data || [];
-    // filter for delete_pattern
-    const activeMatchedReleases = data.filter(
-      ({ draft, tag_name }) => !draft && tag_name.indexOf(deletePattern) !== -1
-    );
-
-    if (activeMatchedReleases.length === 0) {
-      console.log(`😕  no active releases found. exiting...`);
-      return;
-    }
-
-    const matchingLoggingAddition = deletePattern.length > 0 ? " matching" : "";
-
-    console.log(
-      `💬  found total of ${activeMatchedReleases.length}${matchingLoggingAddition} active release(s)`
-    );
-    releaseIdsAndTags = activeMatchedReleases
-      .map(({ id, tag_name: tagName }) => ({ id, tagName }))
-      .slice(keepLatest);
-  } catch (error) {
-    console.error(`🌶  failed to get list of releases <- ${error.message}`);
-    console.error(`exiting...`);
-    process.exitCode = 1;
-    return;
-  }
-
-  if (releaseIdsAndTags.length === 0) {
-    console.error(`😕  no older releases found. exiting...`);
-    return;
-  }
-  console.log(`🍻  found ${releaseIdsAndTags.length} older release(s)`);
-
-  let hasError = false;
-  for (let i = 0; i < releaseIdsAndTags.length; i++) {
-    const { id: releaseId, tagName } = releaseIdsAndTags[i];
-
+  while (true) {
     try {
-      console.log(`starting to delete ${tagName} with id ${releaseId}`);
-
-      const _ = await fetch({
+      let data = await fetch({
         ...commonOpts,
-        path: `/repos/${owner}/${repo}/releases/${releaseId}`,
-        method: "DELETE",
+        path: `/repos/${owner}/${repo}/releases?per_page=100`,
```

Comment:
I think that if the first 100 releases all don't match deletePattern, the action will return prematurely, if the 101th release does match releasePattern. Maybe it's more correct to:
1. building `data` by iterating through the pages until there are no more releases (i.e. `?per_page=100&page=${iteration}`)
2. looping through data to delete the releases (as we already do)

Reply:
I tried to modify it according to your idea, but the author extended the fetch function and only returned the body of the response, so that the data related to the next page of LINK in the api could not be obtained, and it was impossible to determine how much data there was in the end. I'm not particularly familiar with javascript, and I've been testing for a long time, but I haven't been able to succeed. .
I think the focus should be to modify the content of fetch.js so that it returns the response itself, then obtain the LINK data of the response in index.js to determine whether there is data on the next page, and use the loop to continuously obtain data, during which the body of the response is converted to json data.

Reply:
#27